cmd/autoheal/job_runner.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	alertmanager "github.com/openshift/autoheal/pkg/alertmanager"
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal/v1alpha2"
+)
+
+type JobRunnerType int
+
+const (
+	AWX   JobRunnerType = 0
+	BATCH JobRunnerType = 1
+)
+
+type JobRunner interface {
```

Comment:
I am not 100% sure if `JobRunner` is the right name for this interface. Currently it is running either an AWX job or a Kubernetes batch Job, thus the _job_ prefix makes sense. But in general what it runs is a _healing action_, which may be other thing. For example, if in the future we add support for Ansible Runner then it will be running an Ansible Playbook, not a _job_. So consider renaming it to `HealingActionRunner`, or something similar.

Reply:
Sounds good :+1:

Reply:
@nimrodshn I gave the same comment when we talked about this but did't explain it as well...